src/mlpack/core/kernels/pspectrum_string_kernel.hpp

```diff
@@ -73,8 +73,62 @@ class PSpectrumStringKernel
    * @param p The length of substrings to search.
    */
   PSpectrumStringKernel(const std::vector<std::vector<std::string> >& datasets,
-                        const size_t p);
+                        const size_t p) : p(p)
+  {
```

Comment:
This function is a bit long---it might be a bit clearer to move it to `pspectrum_string_kernel_impl.hpp` instead. :+1: